### PR TITLE
Exercise7

### PR DESCRIPTION
**Pull request by jingyophuong**
Our PR review for the group raum9: https://github.com/Systems-Development-and-Frameworks/raum9/pull/8#pullrequestreview-575444299

### CODE REVIEW

**Review by kipmann — APPROVED** [review]

**Review by Titanfood — APPROVED** [review]

File: webapp/components/NavBar.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<template>
+  <nav class="nav-bar">
+    <div>
+      <nuxt-link to="/">Posts</nuxt-link>
+    </div>
+    <div>
+        <nuxt-link to="/login" v-if="!isAuthenticated">Login</nuxt-link>
+        <button v-else @click="logout">Logout</button>
```
Comment (Titanfood) [inline webapp/components/NavBar.vue:8]: Die Test zu der Komponente wurden bereits Bewertet. Die  komponente selbst ist natürlich auch einen ⭐ wert.

**Review by Titanfood — APPROVED** [review]
Innerhalb des Pullrequests gab es keine verschobenen oder Gelöschten Dateien. Bzw. Auch der ⭐ für das korrenkte Ersetzen der Vue-Cli-Webapp durch die nuxt App wurde verdient. Sorry, dass ich 2 mal commitetd habe beim ersten mal sind meine Kommentare verloren gegangen

File: README.md [hunk 2]
```diff
@@ -29,69 +29,16 @@ Cancel: cancel the form submission
...
-npm run dev
-```
-6. Navigate the browser to ```http://localhost:4000/```
-
-7. Once done, the database can be stopped via  
-```
-neo4j stop
-```
+1. Deploy the backend as explained [here](backend/README.md).
+2. Deploy the webapp as explained [here](webapp/README.md).
```
Comment (medizinmensch) [inline README.md:44]: I love that! Not many have done this!

File: backend/README.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+### Installing and running the backend with neo4j in docker
```
Comment (medizinmensch) [inline backend/README.md:1]: The depth of your install instructions are exceptional! 🥇

File: webapp/layouts/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# LAYOUTS
+
+**This directory is not required, you can delete it if you don't want to use it.**
+
+This directory contains your Application Layouts.
+
+More information about the usage of this directory in [the documentation](https://nuxtjs.org/guide/views#layouts).
```
Comment (medizinmensch) [inline webapp/layouts/README.md:7]: Stuff like that could have been hidden with the pre-merge with master. But it's okay.

**Review by medizinmensch — APPROVED** [review]
Hi Team!

Thanks to @Titanfood for already evaluating your PR! That helped!

First of all, I had issues building your app. The Frontend is not starting without the backend (which is not good in itself) and the backend was not starting because of the following error message:

```
> yarn dev
yarn run v1.22.10
$ nodemon -r esm src/index.js
[nodemon] 2.0.7
[nodemon] to restart at any time, enter rs
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting node -r esm src/index.js
/home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:1
GraphQLError: Argument "direction" has invalid value "OUT".
    at getArgumentValues (/home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:240:19)
    at /home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:3059:32
    at Array.forEach (<anonymous>)
    at visitorSelector (/home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:3051:32)
    at callMethod (/home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:2711:24)
    at visitFields (/home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:2853:30)
    at visit (/home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:2781:17)
    at /home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:2769:41
    at Array.forEach (<anonymous>)
    at visit (/home/youri/dev/uni/sdaf/groups/code_party_kp/backend/node_modules/@graphql-tools/utils/index.cjs.js:2762:37) {
  locations: [ { line: 87, column: 125 } ]
}
```

I was able to fix it (by removing the `"` around the `IN`s and `OUT`s), but of course that is not optimal. Anyway, once that was fixed, the test and the site itself worked and your result is great!


⭐ For instructions in the   README.md on how to build your webapp for production.
⭐ For   no changes in "Files Changed" tab of the refactoring from vue-cli   to create-nuxt-app. (See #1 in instructions)
⭐ ⭐ For the API connection   between your front- and backend.
⭐ For   your previous frontend tests still passing. Requests to the backend are   mocked.
⭐ ⭐ For a login feature in your   webapp including a Vue component and its software tests.
⭐ ⭐ For   a menu component which shows a login or logout button and its software tests.
⭐ For an upvote button that   behaves according to the authentication state of your user
⭐ For a   delete and edit button that is only visible to the author of the post.
⭐ For Lighthouse reporting that   your production website is installable as PWA (except HTTPS).
⭐ For   requesting a review and reviewing another team's PR.

It was not easy to get all points in this exercise, but you managed! 
13/13 ⭐

**Pull request merged**